### PR TITLE
[FIX] purchase & stock_dropshipping: fix some tests

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Related to commit https://github.com/odoo/odoo/commit/0e881fcd3a5671c4fe84b2a47c9d3611e4b823a1 and partially adapted in https://github.com/odoo/odoo/commit/6deed451ee686d104583cb33d66c5fbc400cad8d.

@qdp-odoo @nim-odoo @pedrobaeza 

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/purchase/test/fifo_price.yml [hunk 1]
```diff
@@ -4,6 +4,7 @@
   !python {model: res.company}: |
     company = self.browse(cr, uid, ref('base.main_company'))
     company.currency_id = company.env['res.currency'].browse(ref('base.EUR'))
+    cr.execute("UPDATE res_company SET currency_id = %s WHERE id = %s", [company.currency_id.id, company.id])
```
Comment (qdp-odoo) [inline addons/purchase/test/fifo_price.yml]:
```diff
-     company.currency_id = company.env['res.currency'].browse(ref('base.EUR'))
+    cr.execute("UPDATE res_company SET currency_id = %s WHERE id = %s", [ref('base.EUR').id, company.id])
```

Comment (mreficent) [reply addons/purchase/test/fifo_price.yml]: Yeah!

**Comment by mreficent** [comment]
@qdp-odoo fixed, thanks!

**Comment by mreficent** [comment]
@qdp-odoo :green_apple: :green_apple: 

And also I added a similar fix for stock_dropshipping

Ready to merge 😎

**Pull request merged**